pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -312,7 +301,9 @@ private void messageReceived(ConsumerImpl<T> consumer, Message<T> message) {
         // if asyncReceive is waiting : return message to callback without adding to incomingMessages queue
         CompletableFuture<Message<T>> receivedFuture = nextPendingReceive();
         if (receivedFuture != null) {
-            unAckedMessageTracker.add(topicMessage.getMessageId(), topicMessage.getRedeliveryCount());
+            if (listener == null) {
+                unAckedMessageTracker.add(topicMessage.getMessageId(), topicMessage.getRedeliveryCount());
+            }
```

Comment:
seem the listener only use nternalReceive(long timeout, TimeUnit unit), change here seems to be unnecessary